src/containers/screens/main/contracts/Contract.js

```diff
@@ -25,329 +25,363 @@ import ContractInputConstant from '../../../components/contracts/ContractInputCo
  * Screen is used to display the passphrase (mnemonic) of the wallet
  */
 class Contract extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      contractLoaded: false,
-      provider: null,
-      address: '',
-      wallet: this.props.hotWallet.wallet,
-      contractEvents: null,
-      contractFunctions: null,
-      contract: null,
-      withInputs: null,
-      payable: null,
-      functions: [],
-      currentInput: {},
-    };
-  }
+	constructor(props) {
+		super(props);
+		this.state = {
+			// contractLoaded: false,
+			provider: null,
+			address: '',
+			// wallet: this.props.hotWallet.wallet,
+			// contractEvents: null,
+			contractFunctions: null,
+			contract: null,
+			withInputs: null,
+			payable: null,
+			// functions: [],
+			currentInput: {},
+		};
+	}
 
-  componentDidMount = async () => {
-    const provider = await getNetworkProvider(this.props.network);  
-    this.setState({ provider }); 
-  }
+	componentDidMount = async () => {
+		const provider = await getNetworkProvider(this.props.network);
+		this.setState({ provider });
+	}
 
-  getContract = async () => {  
-    const {
-      contractFunctions, contractEvents, contract, withInputs,
-    } = await processContractByAddress(this.state.wallet, this.state.address, this.state.provider, this.props.network);
-    this.setState({
-      contractEvents, contractFunctions, contract, withInputs,
-    });
-  }
+	getContract = async () => {
+		// error handling  
+		const { success, objects } = await processContractByAddress(this.props.hotWallet.wallet,
+			this.state.address, this.state.provider, this.props.network);
+		if (success) {
+			// unused contract events
+			const { contractFunctions, contract, withInputs } = objects;
+			// console.log("in getContract success", contractFunctions, contract, withInputs);
+			this.setState({contractFunctions, contract, withInputs});
+			Toast.show('Success', Toast.LONG);
+		}
+		else {
+			// error in getting contract
+			Toast.show('Error loading contract', Toast.LONG);
+			console.log("in getContract error from contract");
+		}
+	}
 
-  processFunctionInput = (x, inputName, inputType, funcName) => {
-    let c = Object.assign({}, this.state.currentInput);
-    if (c[funcName] == null) {
-      c[funcName] = {};
-    }
-    if (inputType == "string") {
-      c[funcName][inputName] = "'" + x + "'";
-    } else {
-      c[funcName][inputName] = x;
-    }
-    this.setState({ currentInput: c });
-  }
+	processFunctionInput = (text, inputName, inputType, funcName) => {
+		let c = Object.assign({}, this.state.currentInput);
+
+		console.log("in input c: ", c);
+
```

Comment:
@GregTheGreek you can check this log, when putting in inputs for shuffles function.